lib/tuntap.mli

```diff
@@ -60,15 +60,19 @@ val set_up_and_running : string -> unit
     running. Note that when using the [set_ipv4] function, the
     interface will automatically be set up and running. *)
 
-type ipaddr =
-  | AF_INET of Ipaddr.V4.t * Ipaddr.V4.Prefix.t
-  | AF_INET6 of Ipaddr.V6.t * Ipaddr.V6.Prefix.t
+val getifaddrs : ?wanted:([`V4 | `V6] list) -> unit ->
+  [ `V4 of string * Ipaddr.V4.t * Ipaddr.V4.Prefix.t
+  | `V6 of string * Ipaddr.V6.t * Ipaddr.V6.Prefix.t ] list
+(** [getifaddrs ()] is the list that associates interface names with
+    their available IP addresses and their prefix. *)
```

Comment:
can we not have interfaces that have not yet acquired either an IP address of either type?
does this return for all interfaces? wouldn't this be better named `getallifaddrs` or something? and perhaps provide a ```val getifaddrs: ?wanted:([`V4 | `V6] list) -> string -> unit -> [ ... ] list``` as well?